python/ray/tune/examples/pbt_convnet_function_example.py

```diff
@@ -27,36 +29,38 @@ def train_convnet(config, checkpoint_dir=None):
         momentum=config.get("momentum", 0.9),
     )
 
-    # If checkpoint_dir is not None, then we are resuming from a checkpoint.
+    # If `session.get_checkpoint()` is not None, then we are resuming from a checkpoint.
     # Load model state and iteration step from checkpoint.
-    if checkpoint_dir:
+    if session.get_checkpoint():
```

Comment:
```suggestion
    if session.get_checkpoint():
```

Is `get_checkpoint` potentially expensive (does it deserialize something)? Should we have something like `session.has_checkpoint`?

Reply:
Right now, I haven't really scoped the optimization pieces yet (still in the middle of cleaning up APIs). So I would like to defer the decision to later - i.e. if during optimization, we figure that it's better decouple get and has, we can still choose to do so. It won't be a big churn of API.

Reply:
but this is a good point to keep in mind definitely!